manager.go

```diff
@@ -182,38 +184,73 @@ func listenFdFromEnv(ident string) int {
 func (m *manager) handleSignals() {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, syscall.SIGHUP)
-	_ = <-c
 
+	for _ = range c {
```

Comment:
Yeah, I'm targeting 1.2+ at the moment (that's what ships with Ubuntu trusty...), so I need to use the older syntax here.